### PR TITLE
Fixed color clamping

### PR DESCRIPTION
**Pull request by lamarrr**

### CODE REVIEW

File: ZEngine/src/Texture2D.cpp [hunk 1]
```diff
@@ -100,10 +100,10 @@ namespace ZEngine::Rendering::Textures {
 
     void Texture2D::SetData(float r, float g, float b, float a) {
         unsigned char data[5] = {0, 0, 0, 0, '\0'};
-        data[0]               = static_cast<unsigned char>(std::min(r, 255.0f));
-        data[1]               = static_cast<unsigned char>(std::min(g, 255.0f));
-        data[2]               = static_cast<unsigned char>(std::min(b, 255.0f));
-        data[3]               = static_cast<unsigned char>(std::min(a, 255.0f));
+        data[0]               = static_cast<unsigned char>(std::clamp(r, .f, 255.0f));
```
Comment (JeanPhilippeKernel) [inline ZEngine/src/Texture2D.cpp]: you mean `.0f `right?

**Review by JeanPhilippeKernel — CHANGES_REQUESTED** [review]

File: ZEngine/src/Texture2D.cpp (hunk 1 above)
Comment (lamarrr) [reply ZEngine/src/Texture2D.cpp]: Yep

Comment (JeanPhilippeKernel) [reply ZEngine/src/Texture2D.cpp]: compilers want you to be explicit 🙂 , see the pipeline

Comment (lamarrr) [reply ZEngine/src/Texture2D.cpp]: done

**Review by JeanPhilippeKernel — APPROVED** [review]

**Comment by JeanPhilippeKernel** [comment]
Thanks for the fix ! 🙂 @lamarrr

**Comment by lamarrr** [comment]
> Thanks for the fix ! slightly_smiling_face @lamarrr

Sure!

**Pull request merged**